components/eam/src/physics/crm/cospsimulator_intr.F90

```diff
@@ -0,0 +1,3184 @@
+            bounds_name='cosp_htmisr_bnds', bounds=htmisrlim_cosp)
+    end if
+    
+    if (lmodis_sim) then
+       call add_hist_coord('cosp_tau_modis', ntau_cosp_modis,                  &
+            'COSP Mean MODIS optical depth', '1', taumid_cosp_modis,           &
+            bounds_name='cosp_tau_modis_bnds', bounds=taulim_cosp_modis)
+       call add_hist_coord('cosp_reffice',numMODISReffIceBins,                 &
+            'COSP Mean MODIS effective radius (ice)', 'm', reffICE_binCenters_cosp, &
+            bounds_name='cosp_reffice_bnds',bounds=reffICE_binEdges_cosp)
+       call add_hist_coord('cosp_reffliq',numMODISReffLiqBins,                 &
+            'COSP Mean MODIS effective radius (liquid)', 'm', reffLIQ_binCenters_cosp, &
+            bounds_name='cosp_reffliq_bnds',bounds=reffLIQ_binEdges_cosp)      
+    end if
+    
+#endif
+  end subroutine cospsimulator_intr_register
+  
+  ! ######################################################################################
+  ! SUBROUTINE cospsimulator_intr_init
+  ! ######################################################################################
+  subroutine cospsimulator_intr_init()
+
+#ifdef USE_COSP     
+
+    use cam_history,         only: addfld, add_default, horiz_only
+#ifdef SPMD
+    use mpishorthand,        only : mpir8, mpiint, mpicom
+#endif
+    use netcdf,              only : nf90_open, nf90_inq_varid, nf90_get_var, nf90_close, nf90_nowrite
+    use error_messages,      only : handle_ncerr, alloc_err
+    
+    use physics_buffer,  only: pbuf_get_index
+
+    use mod_cosp_config,  only : R_UNDEF    
+    
+    integer :: errcode
+    integer :: i
+
+    ! Are we using MMF?
+    call phys_getopts(use_MMF_out=use_MMF)
+    
+    ! ISCCP OUTPUTS
+    if (lisccp_sim) then
+       !! addfld calls for all
+       call addfld('FISCCP1_COSP',(/'cosp_tau','cosp_prs'/),'A','percent', &
+            'Grid-box fraction covered by each ISCCP D level cloud type',&
+            flag_xyfill=.true., fill_value=R_UNDEF)
+       
+       !*cfMon,cfDa* tclisccp (time,profile), CFMIP wants "gridbox mean cloud cover from ISCCP"
+       call addfld('CLDTOT_ISCCP', horiz_only,'A','percent', &
+            'Total Cloud Fraction Calculated by the ISCCP Simulator ',flag_xyfill=.true., fill_value=R_UNDEF)
+       !*cfMon,cfDa* albisccp (time,profile)
+       ! Per CFMIP request - weight by ISCCP Total Cloud Fraction (divide by CLDTOT_ISSCP in history file to get weighted average)
+       call addfld('MEANCLDALB_ISCCP',horiz_only,'A','1','Mean cloud albedo*CLDTOT_ISCCP',flag_xyfill=.true., fill_value=R_UNDEF)
+       !*cfMon,cfDa* ctpisccp (time,profile)
+       ! Per CFMIP request - weight by ISCCP Total Cloud Fraction (divide by CLDTOT_ISSCP in history file to get weighted average)
+       call addfld('MEANPTOP_ISCCP',horiz_only,'A','Pa','Mean cloud top pressure*CLDTOT_ISCCP',flag_xyfill=.true., &
+            fill_value=R_UNDEF)
```

Comment:
Might be nice to have a "cosp_history" module for this whole section of addfld, and the later outfld calls